luigi/worker.py

```diff
@@ -643,7 +649,9 @@ def _add(self, task, is_complete):
                        resources=task.process_resources(),
                        params=task.to_str_params(),
                        family=task.task_family,
-                       module=task.task_module)
+                       module=task.task_module,
+                       retry_policy_dict=_get_retry_policy_dict(task),
+                       deps_retry_policy_dicts=deps_retry_policy_dicts)
```

Comment:
I saw that dependency tasks are added for the first time with their parent task and it is cashed at this point. Even you send its retry policy to its own task adding method，it won't be updated because it is cached.